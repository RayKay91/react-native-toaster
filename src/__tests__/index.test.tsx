import React from 'react';
import renderer from 'react-test-renderer';
import { Toast } from '..';
import { render, screen } from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';

describe('<Toast />', () => {
  it('renders', async () => {
    const props = {
      title: 'hello',
      subtext: 'world ',
      setToastConfig: jest.fn(),
      setIsVisible: jest.fn(),
      isVisible: false,
      toastQueue: [],
    };
    render(<Toast {...props} />);
    const toast = await screen.getByTestId(/toast/);
    // it is off screen but present in element tree
    expect(toast).toBeOnTheScreen();
  });

  it('does not have visual regressions', () => {
    const props = {
      title: 'hello',
      subtext: 'world ',
      setToastConfig: jest.fn(),
      setIsVisible: jest.fn(),
      isVisible: false,
      toastQueue: [],
    };
    const tree = renderer.create(<Toast {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('is initially off screen', async () => {
    const props = {
      title: 'hello',
      subtext: 'world ',
      setToastConfig: jest.fn(),
      setIsVisible: jest.fn(),
      isVisible: false,
      toastQueue: [],
    };
    render(<Toast {...props} />);
    const toast = await screen.getByTestId(/toast/);
    expect(toast).not.toBeVisible();
  });
});
