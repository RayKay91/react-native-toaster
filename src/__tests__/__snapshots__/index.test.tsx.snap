// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> does not have visual regressions 1`] = `
<View
  collapsable={false}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "borderColor": "#BBB",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 75,
      "opacity": 0,
      "position": "absolute",
      "top": -75,
      "transform": Array [
        Object {
          "translateY": -75,
        },
      ],
      "width": 725,
    }
  }
  testID="toast"
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomLeftRadius": 0,
          "borderRadius": 8,
          "borderTopLeftRadius": 0,
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 10,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
            ]
          }
        >
          hello
        </Text>
      </View>
    </View>
  </View>
</View>
`;
